### PR TITLE
CI/CD: fix packages upgrade, remove unnecessary GH workflow, archive snapcraft logs

### PR DESCRIPTION
**Pull request by marcoppenheimer**

### CODE REVIEW

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -19,4 +19,11 @@ jobs:
       with:
         snap: ${{ steps.build.outputs.snap }}
         release: latest/edge
+    - name: Upload charmcraft logs
+      uses: actions/upload-artifact@v2
```
Comment (welpaolo) [inline .github/workflows/publish.yaml:23]: I think v3 is available

**Review by welpaolo — APPROVED** [review]

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (taurus-forever) [reply .github/workflows/publish.yaml]: Bumped in [1283b0b](https://github.com/canonical/charmed-kafka-snap/pull/16/commits/1283b0beabec538c20cdea00c196111f3f61dc26)

**Comment by taurus-forever** [comment]
I have stabilized tests and build workflows. Please review the PR changes. Tnx!

**Review by taurus-forever — APPROVED** [review]

**Review by welpaolo — APPROVED** [review]
LGTM

**Pull request merged**